crates/codegen/syntax_templates/src/shared/language.rs

```diff
@@ -71,41 +73,66 @@ pub enum ParserResult {
     },
 }
 
+#[derive(Default, Copy, Clone, PartialEq, Eq, Debug, Serialize)]
+pub struct InputPosition {
+    pub byte: usize,
+    pub char: usize,
+}
+
+impl PartialOrd for InputPosition {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.byte.partial_cmp(&other.byte)
```

Comment:
I wonder if we should compare both? for `PartialOrd` and `Ord`.
In an LSP, they can diverge because of edits to documents, while the server still holds symbols/references to both old/new tokens.
It would still be useful to compare them.

Reply:
it must always be the case that comparing the byte offsets and char offsets gives the same Ordering value.

Reply:
👍